src/seismometer/_api.py

```diff
@@ -176,86 +183,126 @@ def fairness_audit(metric_list: Optional[list[str]] = None, fairness_threshold=1
 
+    if fairness_path.exists():
+        return load_as_iframe(fairness_path, height=height)
 
-@export
-@display_cached_widget
-def cohort_selector():
-    """
-    Display an IPyWidget selector to control cohorts used in visualizations.
-    """
-    from seismometer.controls.selection import DisjointSelectionListsWidget
+    target = pdh.event_value(target_column)
+    data = (
+        pdh.event_score(
+            sg.data(),
+            sg.entity_keys,
+            score=sg.output,
+            ref_event=sg.predict_time,
+            aggregation_method=sg.event_aggregation_method(sg.target),
+        )
+        if per_context
+        else sg.data()
+    )
 
-    sg = Seismogram()
-    options = sg.available_cohort_groups
-    widget = DisjointSelectionListsWidget(
-        options=options, value=sg.selected_cohort, title="Select Subgroups of Interest", select_all=True
+    data = data[[target, score_column] + cohort_columns]
+    if len(data.index) < sg.censor_threshold:
+        return template.render_censored_plot_message(sg.censor_threshold)
+    fairness_audit_to_html(
+        fairness_path, data, cohort_columns, score_column, target, score_threshold, metric_list, fairness_threshold
     )
+    return load_as_iframe(fairness_path, height=height)
 
-    def on_widget_value_changed(*args):
-        sg.selected_cohort = widget.value
 
-    widget.observe(on_widget_value_changed, "value")
+# endregion
 
-    # get intial value
-    on_widget_value_changed()
-    return widget
+# region notebook IPWidgets # TODO - move to .controls submodule
 
 
 @export
-@display_cached_widget
 def cohort_list():
     """
     Displays an exhaustive list of available cohorts for analysis.
     """
     sg = Seismogram()
-    from ipywidgets import HBox, Output
+    from ipywidgets import Output, VBox
 
     from .controls.selection import MultiSelectionListWidget
-    from .data.filter import filter_rule_from_cohort_dictionary
 
     options = sg.available_cohort_groups
 
-    comparison_selections = MultiSelectionListWidget(
-        options, ghost_text="Select cohort groups", title="Available Cohorts"
-    )
+    comparison_selections = MultiSelectionListWidget(options, title="Cohort")
     output = Output()
 
     def on_widget_value_changed(*args):
-        output.clear_output()
         with output:
-            if comparison_selections.value:
-                rule = filter_rule_from_cohort_dictionary(comparison_selections.value)
-                cohort_count = sum(rule.mask(sg.dataframe))
-                if cohort_count < 10:
-                    print("Selected Cohort is empty, or has fewer than 10 predictions.")
-                else:
-                    print(f"Selected Cohort has {sum(rule.mask(sg.dataframe))} predictions.")
-            else:
-                print(f"No Cohort Selected: {len(sg.dataframe)} predictions total.")
+            html = _cohort_list_details(comparison_selections.value)
+            output.clear_output()
```

Comment:
Should we clear the output before generating new content? I understand that we want to make the transition seem smooth, but if generation takes a long time it will seem like nothing is happening to the user, which is frustrating. I would argue with a cleared output at least the user knows that the update is in progress.

It really comes down to how long we expect generating new content to be. If its quick then this is fine. Otherwise I think its better to clear first.

Reply:
Added a "Recalculating..." message